extensions/common/gcp/gcp-lib/src/test/java/org/eclipse/dataspaceconnector/gcp/lib/storage/StorageServiceImplTest.java

```diff
@@ -0,0 +1,119 @@
+/*
+ *  Copyright (c) 2022 Google LLC
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Google LCC - Initial implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.gcp.lib.storage;
+
+import com.google.api.gax.paging.Page;
+import com.google.cloud.storage.Blob;
+import com.google.cloud.storage.Bucket;
+import com.google.cloud.storage.BucketInfo;
+import com.google.cloud.storage.Storage;
+import org.eclipse.dataspaceconnector.gcp.lib.common.GcpExtensionException;
+import org.eclipse.dataspaceconnector.spi.monitor.Monitor;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.RETURNS_SMART_NULLS;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class StorageServiceImplTest {
+    private Storage storageMock;
+    private StorageService storageService;
+
+
+    @BeforeEach
+    void setUp() {
+        Monitor monitor = Mockito.mock(Monitor.class);
+        StorageClientFactory clientFactory = Mockito.mock(StorageClientFactory.class, RETURNS_SMART_NULLS);
+        storageMock = Mockito.mock(Storage.class, RETURNS_SMART_NULLS);
+        given(clientFactory.create()).willReturn(storageMock);
```

Comment:
Please stick to established patterns of the code base, i.e. using `when` instead introducing BDDMockito unless there is a good reason not to.